src/uwtools/drivers/driver.py

```diff
@@ -16,103 +11,70 @@
-
-
-    output()
-        Holds the knowledge for how to modify a list of output files and
-        stages them in the working directory.
-
-    job_card()
-        Turns the resources config object into a batch card for the
-        configured Task. Can be called from the command line on the front end
-        node to show the user what the job card would have looked like.
-
-    Notes
-    -------
-    Several functions such as create_model_config() and run() are relegated to
-    specific forecast drivers. This is because they are unique to the app being
-    run and will appropriately parse subsequent stages of the workflow.
+    An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: Optional[str] = None, log: Optional[Logger] = None):
+    def __init__(self, config_file: str, log: Optional[Logger] = None):
         """
-        Initialize the Forecast driver.
+        Initialize the driver.
         """
         self.log = log if log is not None else Logger()
-        if config_file is not None:
-            self.config_file = config_file
-            self._validate()
-        else:
-            self.log.info("No config file provided, available functions are limited.")
+        self._config_file = config_file
+        self._validate()
 
-    def _validate(self) -> bool:
+    # Public methods
+
+    @abstractmethod
+    def batch_script(self) -> None:
         """
-        Validates the objects supplied to the driver from the provided config and platform files.
+        Create a script for submission to the batch scheduler.
         """
-        return config_validator.config_is_valid(
-            config_file=self.config_file,
-            validation_schema=self.schema,
-            log=self.log,
-        )
 
-    @property
     @abstractmethod
-    def schema(self):
+    def output(self) -> None:
         """
-        The schema to validate the config file against.
+        ???
         """
 
     @abstractmethod
-    def requirements(self):
+    def requirements(self) -> None:
         """
-        Recursively parse config and platform files to determine and fill in any dependencies.
+        ???
         """
 
     @abstractmethod
-    def resources(self):
+    def resources(self) -> None:
         """
-        Determine necessary task objects and fill in resource requirements of each.
-
-        Returns a Config object containing the HPC resources needed.
+        ???
         """
 
     @abstractmethod
-    def output(self):
+    def run(self) -> None:
+        """
+        Run the NWP tool.
         """
-        Holds the knowledge for how to modify a list of output files and stages them in the working
-        directory.
 
-        Output files usually are specific to a given app.
+    @abstractmethod
+    def run_cmd(self, *args, run_cmd: str, exec_name: str) -> str:
+        """
+        The command-line command to run the NWP tool.
```

Comment:
I added `run()` and `run_cmd()` to the set of abstract methods. These were already implemented for the `FV3Forecast` driver, but would we have a use case of a `Driver` subclass that would **not** need these methods?

Reply:
I went back to the original PR #214 and I think the intent was to include them as abstract methods but to not assume  they could be defined fully in the base class. Which is to say, I missed making sure they were in the base!